doc/source/f2py/signature-file.rst

```diff
@@ -234,172 +241,188 @@ Other statements:
+  the default. ``required`` should only be specified if there is a need to
+  disable the automatic ``optional`` setting when ``<init_expr>`` is used.
 
-  If Python ``None`` object is used as a required argument, the
+  If a Python ``None`` object is used as a required argument, the
   argument is treated as optional. That is, in the case of array
-  argument, the memory is allocated. And if ``<init_expr>`` is given,
-  the corresponding initialization is carried out.
+  argument, the memory is allocated. If ``<init_expr>`` is given, then the
+  corresponding initialization is carried out.
 
 ``dimension(<arrayspec>)``
-  The corresponding variable is considered as an array with given
-  dimensions in ``<arrayspec>``.
+  The corresponding variable is considered as an array with dimensions given in
+  ``<arrayspec>``.
 
 ``intent(<intentspec>)``
   This specifies the "intention" of the corresponding
   argument. ``<intentspec>`` is a comma separated list of the
   following keys:
 
-  + ``in``
-      The argument is considered as an input-only argument. It means
-      that the value of the argument is passed to Fortran/C function and
-      that function is expected not to change the value of an argument.
-
-  + ``inout``
-      The argument is considered as an input/output or *in situ*
-      output argument. ``intent(inout)`` arguments can be only
-      "contiguous" NumPy arrays with proper type and size.  Here
-      "contiguous" can be either in Fortran or C sense. The latter one
-      coincides with the contiguous concept used in NumPy and is
-      effective only if ``intent(c)`` is used. Fortran contiguity
-      is assumed by default.
-
-      Using ``intent(inout)`` is generally not recommended, use
-      ``intent(in,out)`` instead. See also ``intent(inplace)`` attribute.
-
-  + ``inplace``
-      The argument is considered as an input/output or *in situ*
-      output argument. ``intent(inplace)`` arguments must be
-      NumPy arrays with proper size. If the type of an array is
-      not "proper" or the array is non-contiguous then the array
-      will be changed in-place to fix the type and make it contiguous.
-
-      Using ``intent(inplace)`` is generally not recommended either.
-      For example, when slices have been taken from an
-      ``intent(inplace)`` argument then after in-place changes,
-      slices data pointers may point to unallocated memory area.
-
-  + ``out``
-      The argument is considered as a return variable. It is appended
-      to the ``<returned variables>`` list. Using ``intent(out)``
-      sets ``intent(hide)`` automatically, unless also
-      ``intent(in)`` or ``intent(inout)`` were used.
-
-      By default, returned multidimensional arrays are
-      Fortran-contiguous. If ``intent(c)`` is used, then returned
-      multidimensional arrays are C-contiguous.
-
-  + ``hide``
-      The argument is removed from the list of required or optional
+  * ``in``
```

Comment:
Maybe definition list?

Reply:
This would be doubly nested then (it is already in the definition of `intent`).

![image](https://user-images.githubusercontent.com/4336207/134054415-ecb0e2b8-4bd1-4c09-902d-a2be49906767.png)

I thought the bullet helped rather than having plain whitespace.

Reply:
OK.